### PR TITLE
fix: multi-head-attention incorrect macro usage. @open sesame 1/16 17:11

### LINKED ISSUE #2407: Multi-head attention bug: _MASK_NUM determined at build time, not at run time.
Bug 1

You do NEVER know whether users want FP16 or not in build time; but you assign _MASK_NUM for FP16/FP32 based on preprocessor macro.

https://github.com/nnstreamer/nntrainer/blob/0083caa3f5ebccba2512590e740d056a318e43bc/nntrainer/layers/multi_head_attention_layer.cpp#L497-L509

Assign _MASK_NUM with run-time conditions.

Bug 2

You have muiltiple definition of ```_MASK_NUM``` in this file. @lhs8928 : please look at C/C++ preprocessor references.

### PR DESCRIPTION
**Pull request by myungjoo**
1. Fix re-definitions of macros
2. Determine mask num in runtime, not in compile-time. You do not know whether users want fp16 or not at compile-time.

Fixes #2407

CC: @lhs8928

### CODE REVIEW

**Review by lhs8928 — APPROVED** [review]
LGTM

**Review by skykongkong8 — APPROVED** [review]
LGTM!

File: nntrainer/layers/multi_head_attention_layer.cpp [hunk 1]
```diff
@@ -495,15 +498,10 @@ void MultiHeadAttentionLayer::forwarding(RunLayerContext &context,
...
-#define _MASK_NUM -1e4
-#else
-#define _MASK_NUM -1e10
-#endif
-
   for (unsigned int i = 0; i < mask_dim_height; ++i) {
     for (unsigned int j = i + 1; j < mask_dim_width; ++j) {
-      causal_mask.setValue(0, 0, i, j, _MASK_NUM);
+      causal_mask.setValue(
+        0, 0, i, j, _MASK_NUM(attention_weight.getTensorType().data_type));
```
Comment (djeong20) [inline nntrainer/layers/multi_head_attention_layer.cpp:504]: how about using `getDataType()` for simplicity
```suggestion
        0, 0, i, j, _MASK_NUM(attention_weight.getDataType()));
```

**Review by djeong20 — APPROVED** [review]
LGTM

**Review by SeoHyungjun — APPROVED** [review]
LGTM!

**Pull request merged**